Promote arm64-linux to a non-community triplet

JavierMatosD opened the pull request:
This PR introduces a new CI job for the `arm64-linux` triplet

Commit:
add arm64 linux to CI

BillyONeal commented on scripts/azure-pipelines/linux-arm64/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+FROM ubuntu:jammy-20250404
+
+ARG BUILD_DATE
```
The echo is important here

BillyONeal replied:
Oh, I missed that it's merged into the apt call.

BillyONeal commented on scripts/azure-pipelines/create-docker-images.ps1:
```diff
@@ -67,4 +75,11 @@ if ($BuildLinux) {
+        -Date $Date
+}
+
+docker logout $registry.LoginServer
```
Please add missing trailing newline.

Commit:
we need powershell

Commit:
add newline

Commit:
add autoconf and friends

Commit:
use newer container

Commit:
add missing dependencies

Commit:
update docker

Commit:
add missing X11, OpenGL, image, and math dev libraries to Docker image

Commit:
add missing deps and baseline known failures

Commit:
add python3-venv to image

Commit:
baseline remaining failures

Commit:
Merge remote-tracking branch 'upstream/master' into arm64_ci_workflow

Commit:
format baseline.txt

Commit:
update azure subscription

Commit:
Merge remote-tracking branch 'upstream/master' into arm64_ci_workflow

Commit:
remove passing/unsupported entries

Commit:
format ci.baseline.txt

Commit:
baseline failures

Commit:
Merge remote-tracking branch 'upstream/master' into arm64_ci_workflow

BillyONeal commented:
I marked this draft because @JavierMatosD is OOF at the moment, I'm sure we'll take another look when he's back.

JavierMatosD committed:
merge with master

JavierMatosD committed:
format ci.baseline.txt

JavierMatosD committed:
add newline at the end of file

JavierMatosD committed:
update pool name

JavierMatosD committed:
update arm64-linux image

JavierMatosD committed:
sync ci.baseline.txt with master

JavierMatosD committed:
add --start to sas generation (fix asset cache)

JavierMatosD committed:
update gcc for c++20 features

JavierMatosD committed:
Make aarch64-linux-gnu-gcc point to gcc-12

JavierMatosD committed:
baseline remaining failures

JavierMatosD committed:
format ci.baseline.txt

JavierMatosD committed:
fix typo in ci.baseline.txt

JavierMatosD committed:
baseline xdigest

JavierMatosD commented on scripts/azure-pipelines/linux-arm64/azure-pipelines.yml:
```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation.
```
If I consolidated into one yml, wouldn't I have to duplicate the entire job anyways?

BillyONeal replied:
No, just add a new entry to the outer one, just like how all the Windows ones share.

BillyONeal replied:
https://github.com/JavierMatosD/vcpkg/pull/1 Like this

JavierMatosD committed:
update base image to noble and sync with x64 limux yml

JavierMatosD committed:
baseline remaining failures

JavierMatosD committed:
add dependency for mysql

BillyONeal committed:
Deduplicate the linux yml.

BillyONeal reviewed (dismissed):
I would like to see the deduplication I submitted as https://github.com/JavierMatosD/vcpkg/pull/1

JavierMatosD committed:
Merge pull request #1 from BillyONeal/pr-45331-arm64-linux

BillyONeal committed:
More baseline edits.

* Copy the new failures to `ci.feature.baseline.txt`. (This isn't perfect but better than nothing)
* Add `=pass` entries duplicating `x64-linux` lines.
* Add
    * cctag[cuda]:arm64-linux=feature-fails
    * ctemplate:arm64-linux=fail
    * idevicerestore:arm64-linux=fail
    * ace[xml-utils]:arm64-linux = cascade
    * arrayfire[cuda]:arm64-linux = feature-fails
    * arrayfire[opencl]:arm64-linux = feature-fails
    * arrow[cuda]:x64-linux = feature-fails
    * colmap[cuda,cuda-redist]:arm64-linux = feature-fails
    * flashlight-cuda[fl,imgclass,lm,objdet,asr]:arm64-linux = cascade
     * gz-sim:arm64-linux=skip
     * libfreenect2[openni2]:arm64-linux = cascade
     * libftdi:arm64-linux = cascade
     * libodb-mysql:arm64-linux =
     * libodb-pgsql:arm64-linux = cascade
     * libpmemobj-cpp[benchmark]:arm64-linux = cascade
     * llgl[vulkan]:arm64-linux=feature-fails
     * luasec:arm64-linux = cascade
     * onnx-optimizer[pybind11]:arm64-linux = cascade
     * openni2:arm64-linux = cascade
     * opensubdiv[dx]:arm64-linux = cascade
     * pangolin[openni2]:arm64-linux = cascade
     * pangolin[pybind11]:arm64-linux=feature-fails
     * paraview[cuda]:arm64-linux = feature-fails
     * pcl[openni2,apps]:arm64-linux = cascade
     * qt5[webengine]:arm64-linux = cascade
     * qtmultimedia[gstreamer]:arm64-linux=options
     * sqlitecpp[sqlcipher]:arm64-linux = cascade
     * wpilib:arm64-linux = fail
     * itk[cuda]:arm64-linux = feature-fails
     * opencv4[opengl]:arm64-linux=feature-fails
     * qtmultimedia[gstreamer]:arm64-linux=no-separate-feature-test
     * vtk[core,python,paraview]:arm64-linux = options
     * clapack:arm64-linux=skip
     * vtk-m[cuda]:arm64-linux = skip
* Delete
    * winreg:x64-linux=fail (ports\winreg\vcpkg.json says supports: windows)

JavierMatosD committed:
Merge remote-tracking branch 'upstream/master' into arm64_ci

JavierMatosD committed:
Merge branch 'arm64_ci_workflow' of https://github.com/JavierMatosD/vcpkg into arm64_ci

JavierMatosD committed:
fix ci.feature.baseline

JavierMatosD committed:
remove duplicate entries in ci.feature.baseline.txt

JavierMatosD committed:
exclude arm from gssapi and ldap to resolve vcpkg-ci-curl cascade

JavierMatosD committed:
resolve cmake and cmake-user cascade

JavierMatosD committed:
bump dbus port version

JavierMatosD committed:
version db

JavierMatosD committed:
baseline dbus for arm64-linux and disable 'widgets' feature on arm64-linux

JavierMatosD committed:
format-feature-baseline

JavierMatosD committed:
format-feature-baseline

JavierMatosD committed:
baseline cascades

BillyONeal committed:
Fix feature baseline.

Removes breaks in ci.feature.baseline.txt to make entries for the same port sort together.

[xapian] Fix missing libuuid dependency.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into baseline-fixes

# Conflicts:
#	scripts/ci.baseline.txt
#	scripts/ci.feature.baseline.txt

JavierMatosD committed:
Merge pull request #2 from BillyONeal/baseline-fixes

Fix feature baseline + master merge.

BillyONeal committed:
Fix more baseline cascades and fix feature baseline entries accidentally added to ci.baseline.txt.

BillyONeal committed:
format-feature-baseline

JavierMatosD committed:
Merge pull request #3 from BillyONeal/baseline-fixes

Fix more baseline cascades and fix feature baseline entries accidentally added to ci.baseline.txt.

BillyONeal committed:
Actually promote the triplet,

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	scripts/ci.baseline.txt

BillyONeal committed:
Fix sort in ci.baseline.txt too.

BillyONeal committed:
Record PR build detected failures.

BillyONeal committed:
Move new fails out of the pass block.
